Core/Transaction.cs

```diff
@@ -25,11 +26,31 @@ public Transaction(string cardNumber, Date date, Money amount, string descriptio
         public string Description { get; }
         public string Category { get; }
 
-        public int CompareTo(Transaction other) => Date.CompareTo(other.Date) is var dateComparison && dateComparison == 0 ? GetHashCode().CompareTo(other.GetHashCode()) : dateComparison;
+        private int Hash { get; }
+
+        public int CompareTo(Transaction other) => Date.CompareTo(other.Date);
 
         public override int GetHashCode()
         {
-            return Date.GetHashCode() + Category.GetHashCode() + Amount.GetHashCode() + CardNumber.GetHashCode() + Description.GetHashCode();
+            return Hash;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is Transaction other
+                ? Date == other.Date && Hash == other.Hash
+                : false;
```

Comment:
Codacy found an issue: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=6066087)